_includes/markdown/JavaScript.md

```diff
@@ -237,6 +237,55 @@ module('.element-selector', {
 });
 ```
 
+### Avoid Barrel Files
+
+When you have a large number of files, it can appear useful to create a barrel file to export all of the files in a directory, however this pattern is **not recommended**.
```

Comment:
Have you thought about authoring libraries? Libraries need a way to expose the exports, and if not using barrel files, we would want to recommend using package.json exports. It would be good to touch a bit on it,

Here's an example: https://github.com/10up/headstartwp/pull/679/files#diff-10d255e20438f18796e2a22be13bd7a1cb312bbf1bbbe31444fb480157beb8c9R9

Reply:
@nicholasio Added below